sequencer/src/block/queryable.rs

```diff
@@ -64,6 +64,51 @@ impl QueryablePayload for Payload<TxTableEntryWord> {
         TxIterator::new(ns_table, self)
     }
 
+    fn transaction(
+        &self,
+        meta: &Self::Metadata,
+        index: &Self::TransactionIndex,
+    ) -> Option<Self::Transaction> {
+        let index_usize = index.tx_idx; // TODO fix in https://github.com/EspressoSystems/espresso-sequencer/issues/1010
+        if index_usize >= self.len(meta) {
+            return None; // error: index out of bounds
+        }
+
+        // start
+        let tx_table_range_start = if index_usize == 0 {
+            None
+        } else {
+            let range_proof_start = index_usize.checked_mul(TxTableEntry::byte_len())?;
+            Some(TxTableEntry::from_bytes(self.raw_payload.get(
+                range_proof_start..range_proof_start.checked_add(TxTableEntry::byte_len())?,
+            )?)?)
+        };
+
+        // end
+        let tx_table_range_proof_end = index_usize
+            .checked_add(2)?
+            .checked_mul(TxTableEntry::byte_len())?;
+        let tx_table_range_end = TxTableEntry::from_bytes(self.raw_payload.get(
+            tx_table_range_proof_end.checked_sub(TxTableEntry::byte_len())?
+                ..tx_table_range_proof_end,
+        )?)?;
+
+        let tx_payload_range = tx_payload_range(
+            &tx_table_range_start,
+            &tx_table_range_end,
+            &self.get_tx_table_len(),
+            self.raw_payload.len(),
+        )?;
+        Some(
+            // TODO don't copy the tx bytes into the return value
+            // https://github.com/EspressoSystems/hotshot-query-service/issues/267
+            Transaction::new(
+                crate::VmId(0),
```

Comment:
This seems incorrect. Shouldn't we use `index.ns_idx` and look up the VM ID in the namespace table. This may also explain why we've been seeing more duplicate transactions lately, since the hash of the transaction returned here isn't going to match the hash of the submitted transaction, if it didn't originally have namespace 0.

Reply:
Ok. I thought we were looking for a quick fix to support transactions without computing the proofs but which is consistent with `transaction_with_proof`. 
The actual method is broken (issue here https://github.com/EspressoSystems/espresso-sequencer/issues/1010) but maybe we should make it higher priority to fix this.

Reply:
Right. I can start right away on getting the new transaction lookups in good shape. And then port the changes to `transaction_with_proof` later. Can continue on this PR or start another after this is merged - whatever works best(?)

Reply:
I don't really have a preference, whatever you prefer

Reply:
The namespace support has continued in this follow-up PR: https://github.com/EspressoSystems/espresso-sequencer/pull/1205